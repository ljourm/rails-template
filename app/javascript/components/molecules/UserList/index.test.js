import { shallowMount } from "@vue/test-utils"
import Component from "./"

import axios from "axios"
jest.mock("axios")

const email = "test@example.com"
const res = { data: { users: [{ id: 1, email: email }] } }
axios.get.mockResolvedValue(res)

describe("Component", () => {
  const wrapper = shallowMount(Component)

  test("snapshot", () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  test("emailが表示されること", () => {
    expect(wrapper.find(`emaillink-stub[email="${email}"]`).exists()).toBe(true)
  })
})
