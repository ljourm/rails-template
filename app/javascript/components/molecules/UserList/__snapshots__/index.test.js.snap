// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component snapshot 1`] = `
<div
  class="user-list"
>
  <h2>
    user-list
  </h2>
  <span>
    <emaillink-stub
      email="test@example.com"
    />
  </span>
</div>
`;
